admin-ui/src/main/java/io/crate/rest/action/admin/AdminUIFrontpageAction.java

```diff
@@ -24,29 +24,70 @@
 
 package io.crate.rest.action.admin;
 
+import com.google.common.collect.ImmutableSet;
+import io.crate.rest.CrateRestMainAction;
 import org.elasticsearch.client.Client;
 import org.elasticsearch.common.inject.Inject;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.rest.*;
 
+import java.util.Set;
+
 import static org.elasticsearch.rest.RestRequest.Method.GET;
 
 /**
  * RestHandlerAction to return a html page containing informations about crate
  */
 public class AdminUIFrontpageAction extends BaseRestHandler {
+    private final CrateRestMainAction crateRestMainAction;
+    private final RestController controller;
+    public static final Set<String> USER_AGENTS = ImmutableSet.of(
+        "Mozilla/6.0",
+        "Mozilla/5.0",
+        "Mozilla/4.0",
+        "Mozilla/3.0",
+        "Opera"
+        );
 
     @Inject
-    public AdminUIFrontpageAction(Settings settings, Client client, RestController controller) {
+    public AdminUIFrontpageAction(CrateRestMainAction crateRestMainAction, Settings settings, Client client, RestController controller) {
         super(settings, controller, client);
+        this.crateRestMainAction = crateRestMainAction;
+        this.controller = controller;
+    }
+
+    public void registerHandler() {
+        controller.registerHandler(GET, "/", this);
         controller.registerHandler(GET, "/admin", this);
     }
 
     @Override
     protected void handleRequest(RestRequest request, RestChannel channel, Client client) throws Exception {
+        if (!isBrowser(request.header("user-agent"))){
+            crateRestMainAction.handleRequest(request, channel, client);
+            return;
+        }
+
+        if (request.header("accept") != null &&  request.header("accept").toString().contains("application/json")){
+            crateRestMainAction.handleRequest(request, channel, client);
+            return;
+        }
+
         BytesRestResponse resp = new BytesRestResponse(RestStatus.TEMPORARY_REDIRECT);
         resp.addHeader("Location", "/_plugin/crate-admin/");
         channel.sendResponse(resp);
     }
 
+    private boolean isBrowser(String headerValue) {
+        if (headerValue == null){
+            return false;
+        }
+        for (String user_agent : USER_AGENTS) {
```

Comment:
I think is better to get the first string of user-agent by splitting with whitespaces
and then check if it's contained in the USER_AGENTS set with .contains();